docs: GNU symbol visibility option (#994)
scikit-build/scikit-build: docs/usage.rst

will not explicitly reconfigure the project. Instead, the generated build-system will automatically
detect the change and reconfigure the project after :meth:`skbuild.cmaker.CMaker.make` is called.

Controlling exported symbol visibility
--------------------------------------

> 

Reviewer: Describe what is the default behavior

When using a GNU based compiler on Linux, scikit-build will reduce the binary size by removing all the native exported methods.

> By default, only the `PyInit_<moduleName>` or `init<moduleName>` entry point is exported from the native module.

Reviewer: ```suggestion
By default, only the `PyInit_<moduleName>` (or `init<moduleName>`) entry point is exported from the native module.

> By default, only the ``PyInit_<moduleName>`` (or ``init<moduleName>``)entry point is exported from the native module.

Reviewer: ```suggestion
By default, only the ``PyInit_<moduleName>`` (or ``init<moduleName>``) entry point is exported from the native module.
```

> If you need to preserve native exports in your modle, you may use the `SKBUILD_GNU_SKIP_LOCAL_SYMBOL_EXPORT_OVERRIDE` option as a

Reviewer: ```suggestion
If you need to preserve native exports in your module, you may set the `SKBUILD_GNU_SKIP_LOCAL_SYMBOL_EXPORT_OVERRIDE` as a
```

`CMake option <https://scikit-build.readthedocs.io/en/latest/usage.html#cmake-configure-options>`_
or
`setup.py option <https://scikit-build.readthedocs.io/en/latest/usage.html#scikit-build-options>`_.

> 

Reviewer: ```suggestion

Note that for Python >= 3.9, setting the option `SKBUILD_GNU_SKIP_LOCAL_SYMBOL_EXPORT_OVERRIDE` to `ON` allow to ...
```

It would be nice to further improve to provide some historical contest based on the summary available at SKBUILD_GNU_SKIP_LOCAL_SYMBOL_EXPORT_OVERRIDE

Author: I added a little more detail as to how this works and on what systems.

will not explicitly reconfigure the project. Instead, the generated build-system will automatically
detect the change and reconfigure the project after :meth:`skbuild.cmaker.CMaker.make` is called.

.. _symbol_visibility:

Controlling exported symbol visibility
--------------------------------------

> When using a GNU based compiler on Linux, scikit-build will reduce the binary size by removing all the native exported methods.

Author: @jcfr is there another caveot I am missing as to when this would be used?
